presto-common/src/main/java/com/facebook/presto/common/block/ArrayBlock.java

```diff
@@ -258,8 +258,6 @@ public boolean equals(Object obj)
                 Arrays.equals(this.valueIsNull, other.valueIsNull) &&
                 Objects.equals(this.values, other.values) &&
                 Arrays.equals(this.offsets, other.offsets) &&
-                this.sizeInBytes == other.sizeInBytes &&
-                this.logicalSizeInBytes == other.logicalSizeInBytes &&
```

Comment:
Can we add comments in the method to explain why we moved these two? Same for other classes